components/ExtensionSelector.vue

```diff
@@ -29,18 +32,29 @@
   })
   const { variable_to_update, variable_to_increment } = props
 
+  const loading = ref(true)
   const file_extensions = ref([])
 
+  const toggle_loading = useToggle(loading)
+
   async function get_output_file_extensions() {
     const params = new FormData()
     params.append("geode_object", geode_object)
+    loading.value = true
     await api_fetch(
```

Comment:
If you await, you can just toggle back on the next instruction instead of in each interceptors

Reply:
True that